extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/runtime/SessionOperations.java

```diff
@@ -170,23 +174,30 @@ public static Uni<Mutiny.Session> getSession() {
      * @return the current reactive session stored in the context, or {@code null} if no session exists
      */
     public static Mutiny.Session getCurrentSession() {
-        Context context = vertxContext();
-        Mutiny.Session current = context.getLocal(getSessionKey());
+        Mutiny.Session current = SessionOperations.<Session> contextualDataMap(vertxContext()).get(getSessionKey());
         if (current != null && current.isOpen()) {
             return current;
         }
         return null;
     }
 
+    private static <T> Map<Key<T>, T> contextualDataMap(ContextInternal vertxContext) {
```

Comment:
Instead of duplicating this method here, we could make it public in Hibernate Reactive and keep the `ContextualDataStorage.CONTEXTUAL_DATA_KEY` private.

By doing so, the implementation details about how the `ConcurrentMap` is created/retrieved remains hidden, yet the consumers can manipulate the data it holds.

Reply:
Yes, I will do it